pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -616,7 +643,9 @@ func (r *ReconcileZookeeperCluster) reconcileClusterStatus(instance *zookeeperv1
 	instance.Status.Members.Unready = unreadyMembers
 
 	//If Cluster is in a ready state...
-	if instance.Spec.Replicas == instance.Status.ReadyReplicas && (!instance.Status.MetaRootCreated) {
+	// instance.Spec.Replicas is just an expected value that we set it, but it maybe not take effect by k8s.
+	// So we should check that instance.Status.Replicas is equal to ReadyReplicas, which means true true status of pods.
+	if instance.Status.Replicas == instance.Status.ReadyReplicas && (!instance.Status.MetaRootCreated) {
```

Comment:
If we wont compare with `spec.Replicas` status wont be shown correctly rt, in the case of scaling